src/utils/utils.ts

```diff
@@ -69,3 +69,83 @@ export const createDatesArray = (
   }
   return datesArray
 }
+
+interface CleanedEarn {
+  _id: number
+  data: [
+    {
+      minFee: number
+      maxFee: number
+      dayCount: number
+      memCount: number
+      minDelay: number
+      maxDelay: number
+      minMinutes: number
+      maxMinutes: number
+    }
+  ]
+}
+
+interface EarnAllInfo {
+  _id: number
+  data: [
+    {
+      minFee: number
+      maxFee: number
+      dayCount: number
+      memCount: number
+      minDelay: number
+      maxDelay: number
+      minMinutes: number
+      maxMinutes: number
+    }
+  ]
+  sortedData: {
+    zeroToOne: number
+    oneToTwo: number
+    twoToThree: number
+    threeToTen: number
+  }
+}
+
+export const earnDataParser = (
+  preParsedData: CleanedEarn
+): EarnAllInfo | Error => {
+  let zeroToOne
+  let oneToTwo
+  let twoToThree
+  let threeToTen
+  preParsedData.data.forEach(feeObject => {
+    if (feeObject.minDelay === 0 && feeObject.maxDelay === 1) {
+      zeroToOne = feeObject.maxFee
+    } else if (feeObject.minDelay === 1 && feeObject.maxDelay === 2) {
+      oneToTwo = feeObject.maxFee
+    } else if (
+      feeObject.minDelay === 2 &&
+      feeObject.maxDelay === 3 &&
+      feeObject.maxMinutes === 70
```

Comment:
Noticed this `else if` block was the only one that required `maxMinutes`. Why is that?

Reply:
I wanted to get fees when the minDelay is 2 and the maxDelay is 3, but the API returns two of those specific situations where the maxMinutes are different.  I only wanted one so I added the third parameter to the conditional.